R/applyChangeMeta.R

```diff
@@ -187,29 +187,33 @@ recode_labels <- function(labels, changeTable, existingMeta) {
     single_simpleChanges <- simpleChanges[simpleChanges$varName == var_name, ]
 
     ## Deal with meta data conflicts:
-    # if values will be recoded into each other, do not raise meta data conflict!
-    existing_value_vec <- !is.na(single_simpleChanges$value_new) & single_simpleChanges$value_new %in% single_labels$value
-    values_to_be_recoded <- labels[single_simpleChanges[, "row_num_in_labels"], "value"]
-    existing_value_vec <- existing_value_vec & !single_simpleChanges$value_new %in% values_to_be_recoded
+    values_being_recoded <- labels[single_simpleChanges[, "row_num_in_labels"], "value"]
+
+    # relevant conflicts: existing values that are not themselves being recoded
+    existing_value_logical <- !is.na(single_simpleChanges$value_new) &
+      single_simpleChanges$value_new %in% single_labels$value &
+      !single_simpleChanges$value_new %in% values_being_recoded
+    existing_value_vec <- single_simpleChanges$value_new[existing_value_logical]

```

Comment:
Nit: As this is here defined, I would use it more often 🤔

It might be used instead of `single_simpleChanges[existing_value_logical, "value_new"]` which would reduce, e.g.:

```r
only_new_recode_values <- recode_values[!recode_values %in% single_simpleChanges[existing_value_logical, "value_new"]]
```

to:

```r
only_new_recode_values <- recode_values[!recode_values %in% existing_value_vec]
```
To have it more flexible, it could just be: `existing_value_vec <- single_simpleChanges[existing_value_vec, ]` and when needed, it could be called `existing_value_vec$value_new` or `existing_value_vec$value`. To my opinion, this would reduce complexity in the remaining code. This potential change might mostly address the changes associated with this PR - that's why I noticed it.

Reply:
Big thanks for spotting this; implemented as suggested! 

I think there is further optimization potential, but I will tackle this as part of #59.